msrest/serialization.py

```diff
@@ -478,7 +472,7 @@ def _serialize(self, target_obj, data_type=None, **kwargs):
                 try:
                     ### Extract sub-data to serialize from model ###
                     orig_attr = getattr(target_obj, attr)
-                    if target_obj.is_xml_model():
+                    if is_xml_model_serialization or target_obj.is_xml_model():
```

Comment:
`if is_xml_model_serialization` should be enough, since it will take care of checking `is_xml_model` as fallback.